### ISSUE TITLE
Barcode doesn't provide rawData as a byte array!~

### ISSUE THREAD

**Issue by 815vojta** [googlesamples/android-vision#156]
I use `com.google.android.gms.vision.barcode.BarcodeDetector` in order to read QR codes. The problem is these codes carry raw binary data; no phone numbers, no URLs, no geopoints... I cannot change these QR codes, because they are generated in a different application.

When I scan these QR codes using Android Barcode API, the resulting `Barcode` object doesn't allow me to access the raw data, although it must know it!

All I get is `"Unknown encoding"` message both in `rawValue` and `displayValue` properties.

Having a `byte[]` property instead of two `String` properties would make more sense, since QR codes can carry general data.

I don't see the point of hiding raw data byte array from developers. Could I somehow extend the `BarcodeDetector` class or the `Barcode` class to get what I need?

Thanks a lot!

[Related Stack Overflow question](http://stackoverflow.com/q/40789386/3899583)

**Comment by blindgaenger** [comment]
Yep, I've got a similar use-case with custom aztec codes. Like @815vojta said it would be helpful to provide the raw bytes instead of an error message.

**Comment by pchx** [comment]
Thanks for the feedback.

Unfortunately, there is no way to obtain the raw bytes without adding a new byte[] field since Java requires that Strings contain valid UTF-16. Right now your best option, if you control both the sender and receiver, is to encode your data in a supported character set (e.g. ISO-8859-1).

**Comment by 815vojta** [comment]
Well, the problem is ISO-8859-1 doesn't work. Please, see the related
Stack Overflow question, there you can see my QR code of mode "byte"
encoded using ISO-8859-1, which is valid, but unreadable using this
Library. All I get is "Unknown encoding" in both rawValue and
displayValue, because the byte array wasn't a valid UTF-8 string.
However, the default encoding for QR codes should be ISO-8859-1, not UTF-8.

Vojta

Dne 30.11.2016 v 16:26 pchx napsal(a):

**Comment by 815vojta** [comment]
Is there any progress on this topic? Why couldn't you just add a new byte[] field?

Could you please have a look why this code cannot be decoded?

http://stackoverflow.com/questions/40789386/android-vision-barcode-api-read-custom-qr-code-with-binary-data

**Comment by bsoja** [comment]
+1

**Comment by Wayn0** [comment]
+1

**Comment by kazougaghwaltz** [comment]
+1

**Comment by kotlindroider** [comment]
+1

**Comment by kazougaghwaltz** [comment]
+1

**Comment by Ke4nuReeves** [comment]
+1

**Comment by rchinoy** [comment]
+1

This is a rather large limitation that needs to be addressed.

**Comment by mfranzke** [comment]
+1
